reflect call embed ptr method error

```
package main

import (
	"fmt"
	"reflect"
)

func main() {
	v := reflect.New(reflect.TypeOf(Point2{})).Elem()
	v.Field(0).Set(reflect.New(reflect.TypeOf(Point{})))
	m := v.MethodByName("Set")
	m.Call([]reflect.Value{reflect.ValueOf(100), reflect.ValueOf(200)})
	fmt.Println(v.Interface().(Point2))
}

type Point struct {
	X int
	Y int
}

func (i Point) String() string {
	return fmt.Sprintf("(%v,%v)", i.X, i.Y)
}

func (i *Point) Set(x, y int) {
	i.X, i.Y = x, y
}

type Point2 struct {
	*Point
}
```

llgo run .
```
panic: runtime error: invalid memory address or nil pointer dereference

goroutine 1 [running]:
main.(*Point2).Set(...)
```

--- a/.github/workflows/test_demo.sh
+++ b/.github/workflows/test_demo.sh
@@ -106,6 +106,7 @@ ignore_esp32=(
   "./_demo/go/reflectchanof" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectconv" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectcopy" # panic: internal/bytealg selected .s files require plan9asm translation
+  "./_demo/go/reflectembed" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectempty" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectfunc" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectfnconv" # panic: internal/bytealg selected .s files require plan9asm translation
@@ -197,6 +198,7 @@ ignore_esp32c3_basic=(
   "./_demo/go/reflectifacecall" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectindirect" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectcopy" # panic: internal/bytealg selected .s files require plan9asm translation
+  "./_demo/go/reflectembed" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectempty" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectmethod" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectmake" # panic: internal/bytealg selected .s files require plan9asm translation

--- a/_demo/go/reflectembed/main.go
+++ b/_demo/go/reflectembed/main.go
@@ -1,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func main() {
+	p1 := Point1{&Point{10, 20}}
+	testv(p1)
+	testv(&p1)
+	p2 := Point2{&Point{10, 20}, 0}
+	testv(p2)
+	testv(&p2)
+	p3 := &Point3{Point{10, 20}}
+	testp(p3)
+}
+
+func testv(a any) {
+	v := reflect.ValueOf(a)
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+	m := v.MethodByName("Set")
+	m.Call([]reflect.Value{reflect.ValueOf(100), reflect.ValueOf(200)})
+	if fmt.Sprint(v.Interface()) != "(100,200)" {
+		panic(fmt.Errorf("error: %#v", a))
+	}
+}
+
+func testp(a any) {
+	v := reflect.ValueOf(a)
+	m := v.MethodByName("Set")
+	m.Call([]reflect.Value{reflect.ValueOf(100), reflect.ValueOf(200)})
+	if fmt.Sprint(v.Interface()) != "(100,200)" {
+		panic(fmt.Errorf("error: %#v", a))
+	}
+}
+
+type Point struct {
+	X int
+	Y int
+}
+
+func (i Point) String() string {
+	return fmt.Sprintf("(%v,%v)", i.X, i.Y)
+}
+
+func (i *Point) Set(x int, y int) {
+	i.X, i.Y = x, y
+}
+
+type Point1 struct {
+	*Point
+}
+
+type Point2 struct {
+	*Point
+	n int
+}
+
+type Point3 struct {
+	Point
+}

--- a/runtime/internal/clite/ffi/_wrap/libffi.c
+++ b/runtime/internal/clite/ffi/_wrap/libffi.c
@@ -1,5 +1,5 @@
 #include <ffi.h>
 
-void *llog_ffi_closure_alloc(void **code) {
+void *llgo_ffi_closure_alloc(void **code) {
     return ffi_closure_alloc(sizeof(ffi_closure), code);
 }

--- a/runtime/internal/clite/ffi/ffi_link.go
+++ b/runtime/internal/clite/ffi/ffi_link.go
@@ -44,7 +44,7 @@ func Call(cif *Cif, fn unsafe.Pointer, rvalue unsafe.Pointer, avalue *unsafe.Poi
 
 // void *ffi_closure_alloc (size_t size, void **code);
 //
-//go:linkname ClosureAlloc C.llog_ffi_closure_alloc
+//go:linkname ClosureAlloc C.llgo_ffi_closure_alloc
 func ClosureAlloc(code *unsafe.Pointer) unsafe.Pointer
 
 // void ffi_closure_free (void *);

--- a/runtime/internal/lib/reflect/value.go
+++ b/runtime/internal/lib/reflect/value.go
@@ -2985,7 +2985,7 @@ func methodReceiver(op string, v Value, methodIndex int) (rcvrtype *abi.Type, t 
 			panic("reflect: " + op + " of method on nil interface value")
 		}
 		rcvrtype = iface.itab.typ
-		fn = unsafe.Pointer(iface.itab.fun[i])
+		fn = unsafe.Pointer(unsafe.Slice(&iface.itab.fun[0], i+1)[i])
 		t = (*funcType)(unsafe.Pointer(m.Typ_))
 	} else {
 		rcvrtype = v.typ()

--- a/runtime/internal/lib/reflect/value.go
+++ b/runtime/internal/lib/reflect/value.go
@@ -2664,12 +2664,22 @@ func storeRcvr(v Value, p unsafe.Pointer) {
 		// the interface data word becomes the receiver word
 		iface := (*nonEmptyInterface)(v.ptr)
 		*(*unsafe.Pointer)(p) = ifacePtrData(iface)
-	} else if v.flag&flagIndir != 0 && !ifaceIndir(t) {
-		*(*unsafe.Pointer)(p) = *(*unsafe.Pointer)(v.ptr)
-	} else if v.flag&flagIndir == 0 && runtime.DirectIfaceData(t) {
-		*(*unsafe.Pointer)(p) = unsafe.Pointer(&v.ptr)
+	} else if v.flag&flagIndir != 0 {
+		if ifaceIndir(t) {
+			*(*unsafe.Pointer)(p) = v.ptr
+		} else if runtime.DirectIfaceData(t) {
+			*(*unsafe.Pointer)(p) = v.ptr
+		} else {
+			*(*unsafe.Pointer)(p) = *(*unsafe.Pointer)(v.ptr)
+		}
 	} else {
-		*(*unsafe.Pointer)(p) = v.ptr
+		if ifaceIndir(t) {
+			*(*unsafe.Pointer)(p) = *(*unsafe.Pointer)(v.ptr)
+		} else if runtime.DirectIfaceData(t) {
+			*(*unsafe.Pointer)(p) = unsafe.Pointer(&v.ptr)
+		} else {
+			*(*unsafe.Pointer)(p) = v.ptr
+		}
 	}
 }
 

